frontend/src/pages/projects/projectSharing/ProjectSharingTableRow.tsx

```diff
@@ -0,0 +1,172 @@
+import * as React from 'react';
+import { ActionsColumn, Tbody, Td, Tr } from '@patternfly/react-table';
+import {
+  Button,
+  DropdownDirection,
+  Select,
+  SelectOption,
+  SelectVariant,
+  Text,
+} from '@patternfly/react-core';
+import { CheckIcon, TimesIcon } from '@patternfly/react-icons';
+import { RoleBindingKind } from '~/k8sTypes';
+import { castProjectSharingRoleType, formatDate } from './utils';
+import { ProjectSharingRoleType } from './types';
+
+type ProjectSharingTableRowProps = {
+  obj: RoleBindingKind;
+  isEditing: boolean;
+  typeAhead: string[];
+  onCreateOrEditRoleBinding: (
+    name: string,
+    roleType: ProjectSharingRoleType,
+    rolebinding: RoleBindingKind,
+  ) => void;
+  onCancelRoleBindingCreation: () => void;
+  onEditRoleBinding: (rolebinding: RoleBindingKind) => void;
+  onDeleteRoleBinding: (rolebinding: RoleBindingKind) => void;
```

Comment:
FWIW, not that it overly matters one way or another -- you could just name these without the `roleBinding` aspect -- `onChange` `onDelete` `onEdit` `onCancelEdit`... contextual props is totally fine.

Reply:
Ok, let me change it.